bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v3/ImageImpl.java

```diff
@@ -161,14 +168,65 @@ public String getSrcset() {
 
     @Nullable
     @Override
-    public String getWidth() {
-        return width;
+    @JsonIgnore
+    public String[] getBaseImageResolution() {
+        String baseImageAbsoluteUrl = externalizer.publishLink(resourceResolver, super.getSrc());
+        URL url;
+        InputStream in;
+        BufferedImage image = null;
+        try {
+            url = new URL(baseImageAbsoluteUrl);
+            URLConnection connection = url.openConnection();
+            connection.setRequestProperty("User-Agent", "Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10.4; en-US; rv:1.9.2.2) Gecko/20100316 Firefox/3.6.2");
+            in = connection.getInputStream();
+            image = read(in);
+            in.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        if (image != null) {
+            String height = String.valueOf(image.getHeight());
+            String width = String.valueOf(image.getWidth());
+            return new String[] {width, height};
+        } else {
+            return new String[] {null, null};
+        }
+    }
+
+    @Override
+    public boolean isLazyEnabled() {
+        return !currentStyle.get(PN_DESIGN_LAZY_LOADING_ENABLED, false);
+    }
+
+    @Override
+    @JsonIgnore
+    public String getSrcUriTemplate() {
+        return super.getSrcUriTemplate();
```

Comment:
is it needed to call the method from the parent class, as the method doesn't add any other logic?
Same question applies to the below methods.

Reply:
You're right, `null` can be returned.

Reply:
@comanV Did you see the question above?

Reply:
The getSrcUriTemplate() and getWidths() methods need  to call the method from the parent class because their logic is relevant for Image V3 too, they are used in the getSrcset() method and in the unit tests. I agree that is not needed to call the method from the parent class for getAreas(), because we won’t use Image Maps in Image V3. I modified the method to return `null` and deprecated it.